src/main/java/org/threadly/litesockets/NoThreadSocketExecuter.java

```diff
@@ -52,25 +45,20 @@ protected void addThreadExecutorToClient(Client client) {
    * manually.
    */
   public void wakeup() {
-    if(isRunning()) {
-      commonSelector.wakeup();
-    }
-  }
-
-  @Override
-  public void flagNewWrite(Client client) {
-    updateClientOps(client);
+    checkRunning();
+    commonSelector.wakeup();
   }
 
   @Override
-  public void flagNewRead(Client client) {
-    updateClientOps(client);
-  }
-  
-  private void updateClientOps(Client client) {
+  public void setClientOperations(final Client client) {
     ArgumentVerifier.assertNotNull(client, "Client");
-    if(isRunning() && clients.containsKey(client.getChannel())) {
-      if(client.canWrite() && client.canRead()) {
+    if(isRunning() && !clients.containsKey(client.getChannel()) && !client.isClosed() && client.getClientsSocketExecuter() == this) {
```

Comment:
Seems weird how your checking if it contains the key, and doing a putIfAbsent call.  You should just do the putIfAbsent and check to see if it returns an existing reference (and if it does, then you know the put failed).